primefaces/src/main/java/org/primefaces/component/api/UICalendar.java

```diff
@@ -370,12 +376,53 @@ protected void createFacesMessageFromValidationResult(FacesContext context, Vali
     }
 
     public void validateMinMax(FacesContext context) {
-        LocalDate minDate = CalendarUtils.getObjectAsLocalDate(context, this, getMindate());
-        LocalDate maxDate = CalendarUtils.getObjectAsLocalDate(context, this, getMaxdate());
+        Instant minDate = toInstant(context, getMindate(), "minimum");
+        Instant maxDate = toInstant(context, getMaxdate(), "maximum");
         if (minDate != null && maxDate != null && maxDate.compareTo(minDate) < 0) {
             String id = getClientId(context);
             String component = this.getClass().getSimpleName();
             throw new FacesException(component + " : \"" + id + "\" minimum date must be less than maximum date.");
         }
     }
+
+    private Instant toInstant(FacesContext context, Object date, String attr) {
```

Comment:
I wonder if this method should be moved to CalendarUtils?

Reply:
Think so too

Reply:
If `toInstant` has to be moved to `CalendarUtils it should be renamed to somthing like `getObjectAsInstant`. The parameter `attr` (which is only used to construct a helpful FacesException) has to be reworked somewhat. And as in `CalendarUtils.getObjectAsLocalXxx` an additional parameter `UICalendar calendar` has to be added.
If the programming guidelines suggest that this code belongs in a util class, then I'll be happy to rebuild it.
What should the moved method be called?

Reply:
i like getObjectAsInstant as it keeps the pattern the same as getObjectAsLocal...